module4-authentication-and-security/r1-securing-backend-app/README.md

```diff
@@ -0,0 +1,141 @@
+# Securing a backend app
+
+So far, you've learned how to create server side applications and how to use databases to persist and manipulate the data using CRUD operations. All these features sound awesome, but as it stands currently, they come with a high vulnerability and security risk.
+
+In this module, we will go over how we secure our app and how to make sure that neither intended nor unintended security breaches happen.
+
+Follow me in the following journey to understand why we need security.
+
+## Wishlist inventory app
+
+You've created an amazing app to store a shopping wishlist for your users. You have endpoints that retrieve the list, add an item to the list, update an item in the list, and delete an item in the list.
+
+I liked the app and started using it to create 2 lists for the things that I love, computers and kitchenware, and started adding items to both.
+
+In your database, you have 2 tables that store these items as follows:
+
+### Table 1: lists
+
+| id  | name           |
+| --- | -------------- |
+| 1   | Computer Parts |
+| 2   | Kitchenware    |
+
+### Table 2: items
+
+| id  | name                 | list_id | url                   | price | bought |
+| --- | -------------------- | ------- | --------------------- | ----- | ------ |
+| 1   | NVidia RTX 3070      | 1       | https://amzn.com/313  | $785  | false  |
+| 2   | Philips Blender      | 2       | https://url.com/313   | $90   | false  |
+| 3   | AMD Ryzen 9          | 1       | https://amzn.com/585  | $599  | true   |
+| 4   | Tefal Iceforce knife | 2       | https://amzn.com/1311 | $20   | false  |
+
+Now, suppose another user comes to your app: how will your app decide what to show them? When they land on the homepage, they would see my items because the code can't decide what to show. This is the first security risk. The app needs somehow to relate the data to a specific user, so it only queries data that belongs to the requesting user.
+
+That's why almost everywhere online, you need to register or log in to an account in order to use an application, so the data can be related to you and you only.
+
+## Let's add some security
```

Comment:
i feel this should be "Let's add some authentication" -- usually i would think of security as more of securing the app against attacks and authentication as slightly more specific as adding identity related things